Code/GraphMol/SubstructLibrary/substructLibraryTest.cpp

```diff
@@ -692,38 +692,39 @@ void testSegFaultInHolder() {
 void testTautomerQueries() {
   BOOST_LOG(rdErrorLog) << "-------------------------------------" << std::endl;
   BOOST_LOG(rdErrorLog) << "   testTautomerQueries" << std::endl;
-    
+
   boost::shared_ptr<CachedTrustedSmilesMolHolder> mols1(
-							new CachedTrustedSmilesMolHolder());
+      new CachedTrustedSmilesMolHolder());
   mols1->addSmiles("CN1C2=C(C(=O)Nc3ccccc3)C(=O)CCN2c2ccccc21");
   SubstructLibrary sss(mols1);
   auto query = "Cc1nc2ccccc2[nH]1"_smiles;
-  //auto matches1 = sss.getMatches(*query);
-  //TEST_ASSERT(matches1.size() == 0);
-  auto tq = TautomerQuery::fromMol(*query);
```

Comment:
This, of course, would work if we returned unique_ptrs from our apis :)

Reply:
yep. If only we weren't constrained by backwards compatibility.

I actually had a bit of inspiration about how to solve this problem... email coming on that.